src/plugins/upf/upf.c

```diff
@@ -773,6 +773,263 @@ VLIB_CLI_COMMAND (upf_show_session_command, static) =
 };
 /* *INDENT-ON* */
 
+static clib_error_t *
+upf_create_app_command_fn (vlib_main_t * vm,
```

Comment:
don't reinvent the wheel. The nwi CLI functions do something very similar.

Use the upf_nwi_add_del_command_fn and the vnet_upf_nwi_add_del functions as blueprint for this.